client/state/data-layer/http/index.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * External dependencies
+ */
+import superagent from 'superagent';
+
+/***
+ * Internal dependencies
+ */
+import { extendAction } from 'state/utils';
+import { HTTP_REQUEST } from 'state/action-types';
+import { failureMeta, successMeta } from 'state/data-layer/wpcom-http';
+
+/***
+ * Handler to perorm an http request based on `HTTP_REQUEST` action parameters:
+ * {String} url the url to request
+ * {String} method the method we should use in the request: GET, POST etc.
+ * {Array<{Object}>} headers array of { key: '', value: '' } pairs for the request headers
```

Comment:
simply curious, is there a reason to use an object here with `key` and `value` properties vs an array of pairs?

```js
[
	[ 'Authorization', 'Basic: blarg' ],
	[ 'Accept-Encoding', 'application/json' ],
]
```

a list of pairs seems at least plausibly appropriate and would destructure well below…

```js
request = headers.reduce( ( r, [ key, value ] ) => r.set( key, value ), request );
```